com.avaloq.tools.ddk.xtext.scope/src/com/avaloq/tools/ddk/xtext/scope/generator/ScopeProviderGenerator.xtend

```diff
@@ -102,58 +102,80 @@ class ScopeProviderGenerator {
   def scopeMethods(ScopeModel it, String baseName) '''
     @Override
     protected IScope doGetScope(final EObject context, final EReference reference, final String scopeName, final Resource originalResource) {
-      «FOR name : allScopes().filter(s|s.reference !== null).map(s|s.getScopeName()).toSet().sortBy(n|if (n=="scope") "" else n) SEPARATOR " else "
-     »if ("«name»".equals(scopeName)) {
+      if (scopeName == null) {
+        return null;
+      }
+
+      switch (scopeName) {
+      «FOR name : allScopes().filter(s|s.reference !== null).map(s|s.getScopeName()).toSet()
+     »case "«name»":
         «FOR scope : allScopes().filter(s|s.reference !== null).filter(s|s.getScopeName()==name)»
         if (reference == «scope.reference.literalIdentifier()») return «scope.scopeMethodName()»(context, reference, originalResource);
         «ENDFOR»
-      }«
+        break;
+      «
       ENDFOR»
```

Comment:
Can these be combined into a single line, for consistency?

Reply:
done